lib/ddtrace/sync_writer.rb

```diff
@@ -10,9 +10,17 @@ def initialize(options = {})
       end
     end
 
-    def write(trace, services)
+    def write(trace, services = nil)
+      unless services.nil?
+        Datadog::Patcher.do_once('SyncWriter#write') do
```

Comment:
I am not sure this does 100% what I expect it to... seems like it, but /shrug

Reply:
Based on https://github.com/DataDog/dd-trace-rb/blob/42be52940b35ebfca450cf19b294ea83a82785f4/lib/ddtrace/patcher.rb#L24 
It will do the thing in block only once.

Not necessarily an intended use case of this but if threads are of no concern it should be good to go.

Reply:
Yeah, it is definitely the point to only do it once. Mainly because Ruby doesn't have the rate limited logger yet, and if we happened to miss a place in code or someone is doing this manually, then it could get super spammer.

I was mostly concerned about calling this method statically instead of as an instance method.

Reply:
Yeah it should work, however its not thread safe so using it in this particular class might lead to unexpected results. Its only logs so at words they will be set 1 or 2 times more than expected.

Also SyncWriter itself should be highly discourage from use. It has some very unpleasant side effects in multithreaded environment

Reply:
:+1: yeah, as long as we dedupe a bit, even once per-thread is fine by me, as long as it is 1M messages per-thread 😆

Reply:
Generally don't think calling `Datadog::Patcher.do_once` is a good idea; the state is effectively turned into global state, which has some really bad side effects (particularly for things like tests.) The `Datadog::Patcher` module was meant to be composed in, not be called directly.

We should probably compose in `Datadog::Patcher` and use it that way, or extract `do_once` into another module with a better name, then compose that in here.